docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
+
+In javascript
+-------------
+
+The easiest way to use Switchboard in conjunction with Javascript is to set a
+Javascript flag within your template code and then use that flag within your
+Javascript logic. Using Mako's syntax in the template::
+
+    <%!
+        from switchboard import operator
+    %>
+    <script>
+        window.switches = window.switches || {};
+        % if operator.is_active('foo'):
+        switches.foo = true;
+        % else:
+        switches.foo = false;
+        % endif
+    </script>
+
+In the Javascript::
+
+    if (switches.foo) {
+        ... do something ...
+    } else {
+        ... do something else ...
+    }
+
+Again, this time using Jinja syntax and the Switchboard-provided "active"
+test_::
+
+    <script>
+        window.switches = {};
+        switches.foo = {{ 'true' if 'foo' is active else 'false' }};
+    </script>
+
+Context objects
+---------------
+
+Every switch is evaluated (to see if it is active or not) within a particular
+context. By default, that context includes the request object, which allows
+Switchboard to specify conditions such as: "make this switch active only for
+requests with 'foo' in the query string." That said, there may be other
+objects that would be handy to have available in the context. For example, in
+an ecommerce setting, the Product model may have a ``new`` flag. By passing
+the model into the ``is_active`` method, Switchboard can now activate
+switches based on that flag::
+
+    if operator.is_active('foo', my_product):
+
+Any objects passed into the ``is_active`` method after the switch's key will be
+added to the context. Normally when dealing with context objects, a custom
+condition will be required to actually evaluate the switch against that object.
+
+Custom conditions
+-----------------
+
+Switchboard supports custom conditions, allowing application developers to
+adapt switches to their particular needs. Creating a condition typically
+consists of extending `switchboard.conditions.ConditionSet`.
+
+An example: if the application needs to activate switches for visitors from a
+particular country, a custom condition can do the geo lookup on the IP from
+the request and return the country value::
+
+    from switchboard.conditions import ConditionSet, Regex
+    from my_app.geo import country_code_by_addr
+
+    class GeoConditionSet(ConditionSet):
+        countries = Regex()
+
+        def get_namespace(self):
+            return 'geo'
+
+        def get_field_value(self, instance, field_name):
+            if field_name == 'countries':
+                return country_code_by_addr(client_ip())
+
+        def get_group_label(self):
+            return 'Geo'
+
+The first thing in the custom condition is to define the fields that makeup the
+condition. In this case, there is one "countries" field, which is a regex,
+allowing admins to specify criteria like `(US|CA)` (US or Canada). Here are the
+fields supported by Switchboard:
+
```

Comment:
So even though 'countries' is not used in the given code snippet, it is defined for use in the creation of the switch?